### ISSUE TITLE
server-to-server sharing can't edit a single shared file

### ISSUE THREAD

**Issue by schiesbn** [owncloud/core#13617]
This happens on master:

1. I share a single file, let's say "welcome.txt" from user1 at server1 to user2 at server2
2. login as user2 at server2 and accept the share
3. user2 can see the share and also read/download it
4. user2 tries to edit the file with the external text editor.
5. File is not updated in server1

owncloud.log:

```
{"reqId":"8b6ac9eb40b19046372633557cdbee95","remoteAddr":"192.168.1.161","app":"webdav client","message":"curl GET http:\/\/vm\/public.php\/webdav\/ returned status code 400","level":3,"time":"2015-01-22T22:21:49+00:00"}
```

I'm a bit surprised about the "GET", I would expect a "PUT" here.

I also tried to update the file directly with curl at the command line ```curl -T modifiedwelcome.txt http://CzmO5xWquA3WVZH@vm/public.php/webdav/```

but this gave me:

```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\Conflict</s:exception>
  <s:message>Files can only be created as children of collections</s:message>
</d:error>
```

If I share a folder and edit a file inside the shared folder everything works fine.

Any idea what's broken here? I'm sure this worked a few weeks ago. @icewind1991 maybe you have an idea?

**Comment by DeepDiver1975** [comment]
@PVince81 sounds familiar ...

**Comment by PVince81** [comment]
Could be a duplicate of https://github.com/owncloud/core/issues/11722

At least when sharing with link there is no way for the sharer to set the "allow editing" option.

If the "invite workflow" route properly sets the permission then it might be a different issue.

**Comment by schiesbn** [comment]
> At least when sharing with link there is no way for the sharer to set the "allow editing" option.

Yes, but if you share a file with the new server-to-server sharing you can set "can edit" also for files. But maybe this is a limitation of webdav because the we point directly to a file and not to a collection and you probably can't write directly to a webdav end point but only to files within a collection. Does this makes sense?

**Comment by PVince81** [comment]
Hmmm... possibly but not sure. If it's the case we'd need a new endpoint for such cases.
I could imagine having a new style WebDAV endpoint called `/remote.php/publicshares/token/$targetPath`

where `$targetPath` can be either a file or subfolder.

In the case of file share:
`/remote.php/shares/token/file` (GET and PUT this endpoint)

In the case of folder share, token is the collection containing the files inside the remote folder:
`/remote.php/publicshares/token/`

Anyway... might need a short term workaround first.

- [ ] investigate the WebDAV issue first

**Comment by DeepDiver1975** [comment]
I'm getting this exception when testing a single file share:
````
{"reqId":"addc222aa7090ae5d973a26542350d57","remoteAddr":"::1","app":"PHP","message":"Argument 1 passed to OC\\Connector\\Sabre\\ObjectTree::init() must be an instance of Sabre\\DAV\\ICollection, instance of OC_Connector_Sabre_File given, called in \/home\/deepdiver\/Development\/ownCloud\/core\/apps\/files_sharing\/publicwebdav.php on line 67 and defined at \/home\/deepdiver\/Development\/ownCloud\/core\/lib\/private\/connector\/sabre\/objecttree.php#42","level":3,"time":"2015-01-26T15:38:09+00:00","method":"PROPFIND","url":"\/public.php\/webdav\/"}

**Comment by DeepDiver1975** [comment]
https://github.com/owncloud/core/blob/master/apps/files_sharing/publicwebdav.php#L67

**Comment by DeepDiver1975** [comment]
A dummy collection has to be passed in holding exactly this one note ... preparing PR

**Comment by DeepDiver1975** [comment]
this totally falls apart then

![05466030](https://cloud.githubusercontent.com/assets/1005065/5903114/1f4e8cc2-a580-11e4-99b5-bb5486439f14.jpg)

**Comment by MorrisJobke** [comment]
fixed in #13682
